packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -3,107 +3,201 @@ import {
   FarmFilterOptions,
   FilterOptions,
   generateString,
+  GridClient,
   MachineModel,
   MachinesModel,
   NetworkModel,
   randomChoice,
+  TwinDeployment,
 } from "../src";
 import { config, getClient } from "./client_loader";
 import { log } from "./utils";
 
+async function pingNodes(grid3: GridClient, nodes: any[]): Promise<any[]> {
+  const pingPromises = nodes.map(async node => {
+    const pingPromise = grid3.zos.pingNode({ nodeId: node.nodeId });
+    const timeoutPromise = new Promise((_, reject) =>
+      setTimeout(() => reject(new Error(`Timeout: Ping to node ${node.nodeId} took too long`)), 10000),
+    );
+
+    try {
+      const res = await Promise.race([pingPromise, timeoutPromise]);
```

Comment:
instead of promise race, we can set the expiration to 10 sec in `pingNode` method